src/states/node.rs

```diff
@@ -808,6 +832,144 @@ impl Node {
         }
     }
 
+    fn handle_node_approval_vote(&mut self,
+                                 candidate_name: XorName,
+                                 validity: bool)
+                                 -> Evented<Result<(), RoutingError>> {
+        let mut result = Evented::empty();
+        // Once the joining node joined, it may receive the vote regarding itself.
+        // Or a node may receive CandidateApproval before connection established.
+        let (client_auth, peer_id) = try_ev!(
+            self.peer_mgr.handle_node_approval_vote(candidate_name, validity),
+            result);
+
+        if validity {
+            let sections = self.peer_mgr.pub_ids_by_section();
+            info!("{:?} Sending NodeApproval to {:?}.", self, candidate_name);
+            let _ = self.send_routing_message(RoutingMessage {
+                    src: Authority::Section(candidate_name),
+                    dst: client_auth,
+                    content: MessageContent::NodeApproval { sections: sections },
+                })
+                .extract(&mut result);
+        } else {
+            self.disconnect_peer(&peer_id);
+        }
+        result.map(Ok)
+    }
+
+    fn handle_approval_confirmation(&mut self,
+                                    candidate_name: XorName)
+                                    -> Evented<Result<(), RoutingError>> {
+        let mut result = Evented::empty();
+        let (pub_id, peer_id) = try_ev!(self.peer_mgr.handle_approval_confirmation(candidate_name),
+                                        result);
+        self.add_to_routing_table(&pub_id, &peer_id).extract(&mut result);
+        result.map(Ok)
+    }
+
+    fn handle_node_approval(&mut self, sections: &SectionMap) -> Evented<Result<(), RoutingError>> {
+        let mut events = Evented::empty();
+        if self.is_approved {
+            warn!("{:?} Received duplicate NodeApproval.", self);
+            return events.with_value(Ok(()));
+        }
+
+        self.get_approval_timer_token = None;
+
+        self.peer_mgr.add_prefixes(sections.keys().into_iter().map(|&prefix| prefix).collect_vec());
```

Comment:
Also, since this modifies the RT's `Prefix`es, we should add the following:
```rust
trace!("{:?} Node approved. Prefixes: {:?}", self, self.peer_mgr.routing_table().prefixes());
```